File: testSuite/test-galacticStructureStateDeallocateBug.pl
#!/usr/bin/env perl
use strict;
use warnings;
use PDL;
use PDL::NiceSlice;
use PDL::IO::HDF5;

# Run models that test for regression of the deallocate bug in galacticStructureState
# Andrew Benson (02-May-2022)

# Make output directory.
system("mkdir -p outputs/");

# Run the model.
system("cd ..; ./Galacticus.exe testSuite/parameters/galacticStructureStateDeallocateBug.xml");
if ( $? == 0 ) {
    print "SUCCESS:  galacticStructureState deallocate bug\n";
} else {
    print "FAIL: galacticStructureState deallocate bug model failed to run\n";
}

exit;

File: testSuite/test-galacticStructureStateDeallocateBug.py
#!/usr/bin/env python3
import subprocess
import sys

# Test for galactic structure state deallocate bug.
# Andrew Benson (ported to Python)

# Ensure output directory exists.
subprocess.run("mkdir -p outputs", shell=True)

# Run the model.
status = subprocess.run("cd ..; ./Galacticus.exe testSuite/parameters/galacticStructureStateDeallocateBug.xml", shell=True)
if status.returncode != 0:
    print("FAILED: model failed to run")
    sys.exit(0)

print("SUCCESS: galactic structure state deallocate bug test")
